src/Rfc3339.elm

```diff
@@ -237,51 +224,35 @@ dateTimeParser =
                     ]
             , Parser.Advanced.succeed Nothing
             ]
-        |> Parser.Advanced.andThen
-            (\dateTime ->
-                case dateTime of
-                    TimeLocal _ ->
-                        Parser.Advanced.succeed dateTime
-
-                    DateLocal date ->
-                        let
-                            maxDays : Int
-                            maxDays =
-                                daysInMonth date
-                        in
-                        if date.day > maxDays then
-                            Parser.Advanced.problem (PrbDayTooLarge maxDays)
 
-                        else
-                            Parser.Advanced.succeed dateTime
 
-                    DateTimeLocal date ->
-                        let
-                            maxDays : Int
-                            maxDays =
-                                daysInMonth date
-                        in
-                        if date.day > maxDays then
-                            Parser.Advanced.problem (PrbDayTooLarge maxDays)
+fakeZone : { hour : Int, minute : Int } -> Time.Zone
+fakeZone offset =
+    Time.customZone
+        (if offset.hour >= 0 then
+            offset.hour * 60 + offset.minute
 
-                        else
-                            Parser.Advanced.succeed dateTime
+         else
+            offset.hour * 60 - offset.minute
+        )
+        []
 
-                    DateTimeOffset date ->
-                        let
-                            maxDays : Int
-                            maxDays =
-                                daysInMonth date
-                        in
-                        if date.day > maxDays then
-                            Parser.Advanced.problem (PrbDayTooLarge maxDays)
 
-                        else
-                            Parser.Advanced.succeed dateTime
-            )
+checkDay : { a | year : Int, month : Time.Month, day : Int } -> Parser Date.Date
+checkDay date =
```

Comment:
Nice!

Reply:
I know, right??? When I realized I was so happy with it